hack/sdn-kind-cno.sh

```diff
@@ -0,0 +1,239 @@
+BUILD_SDN=${BUILD_SDN:-false}
+BUILD_CNO=${BUILD_CNO:-false}
+BUILD_MULTUS=${BUILD_MULTUS:-false}
+CNO_PATH=${CNO_PATH:-$GOPATH/src/github.com/openshift/cluster-network-operator}
+SDN_PATH=${SDN_PATH:-$GOPATH/src/github.com/openshift/sdn}
+KIND_CONFIG=${KIND_CONFIG:-$HOME/kind-sdn-config.yaml}
+export KUBECONFIG=${HOME}/kube-sdn.conf
+NUM_MASTER_NODES=1
+SDN_KIND_VERBOSITY=${SDN_KIND_VERBOSITY:-5}
+
+# Default networks (same as in KIND)
+IP_FAMILY=ipv4
+CLUSTER_CIDR=${CLUSTER_CIDR:-"10.244.0.0/16"}
+SERVICE_NETWORK=${SERVICE_NETWORK:-"10.96.0.0/12"}
+HOST_PREFIX=24
+
+# Check for docker
+if ! command -v docker; then
+  echo "docker binary missing in PATH. Docker is required for this deployment"
+  exit 1
+fi
+
+# Check for kubectl
+if ! command -v kubectl; then
+  echo "kubectl binary missing in PATH. Please build/install kubectl."
+  exit 1
+fi
+
+# Ensure reachability to host via Docker network
+if ! sudo iptables -C DOCKER-USER -j ACCEPT > /dev/null 2>&1; then 
+  sudo iptables -I DOCKER-USER -j ACCEPT
+fi
+
+ # create the config file
+  cat <<EOF > ${KIND_CONFIG}
+# config for 1 control plane node and 2 workers (necessary for conformance)
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+networking:
+  ipFamily: ipv4
+  disableDefaultCNI: true
+  podSubnet: ${CLUSTER_CIDR}
+  serviceSubnet: ${SERVICE_NETWORK}
+featureGates:
+  APIPriorityAndFairness: true
+runtimeConfig:
+  flowcontrol.apiserver.k8s.io/v1alpha1: true
+nodes:
+- role: control-plane
+- role: worker
+- role: worker
+EOF
+
+
+# Create KIND cluster
+kind create cluster --name sdn --image kindest/node:${K8S_VERSION} --config=${KIND_CONFIG} -v ${SDN_KIND_VERBOSITY}
+#kind create cluster --name sdn --image ${KIND_IMAGE} --config=${KIND_CONFIG} -v ${SDN_KIND_VERBOSITY}
+
+echo -e "\n"
+
+CNO_TEMPLATES=$CNO_PATH/manifests
+
+if [ "$BUILD_CNO" = true ]; then
+  echo "Building CNO"
+  pushd $CNO_PATH
+  # sed -i '/host-run-netns/{n;s/readOnly.*/mountPropagation: Bidirectional/}' bindata/network/ovn-kubernetes/ovnkube-node.yaml
+  CNO_IMAGE=$(BUILDCMD="docker build" ./hack/build-image.sh | grep 'Successfully tagged' | grep -Eo cluster-network-operator:.*)
+  # sed -i '/host-run-netns/{n;s/mountPropagation.*/readOnly: true/}' bindata/network/ovn-kubernetes/ovnkube-node.yaml
+  if [ -z "$CNO_IMAGE" ]; then
+    echo "Error locating built CNO Image"
+    exit 1
+  fi
+  echo "Loading CNO image into KIND"
+  kind load docker-image $CNO_IMAGE --name sdn -v ${SDN_KIND_VERBOSITY}
+  popd
+fi
+
+if [ "$BUILD_SDN" = true ]; then
+  echo "Building SDN"
+  pushd $SDN_PATH
+  make clean
+  #docker build --no-cache -f images/sdn/Dockerfile.fedora -t sdn-test .
+  popd
+  echo "Loading SDN docker image into KIND"
+  kind load docker-image sdn-test --name sdn -v ${SDN_KIND_VERBOSITY}
+fi
+
+OVS_PASSWD=$(docker run -t --rm --entrypoint 'grep' quay.io/openshift/origin-ovn-kubernetes:latest openvswitch /etc/passwd)
+NODES=$(docker ps | grep "kindest/node" | awk '{ print $1 }')
```

Comment:
Could this grep on "sdn"? I've used this script and I had two clusters running ("ovn" and "sdn") and this led to stuff accidentally being done on my "ovn" cluster

Reply:
we should use kind api
```
NODE=$(kind get nodes --name ${KIND_CLUSTER_NAME})
```

Reply:
You're right, I've changed it for:
`NODES=$(docker ps --filter label='io.x-k8s.kind.cluster=sdn' --format '{{.Names}}')`

Which should be far more reliable.

Reply:
Oops just read you Antonio, will do that instead, sounds a bit more reliable

Reply:
it´s a wrapper of your command :smile: